scripts/ci/build-and-test.sh

```diff
@@ -10,20 +10,29 @@ cd $(dirname $0)/../..
 source scripts/ci/sources/mode.sh
 source scripts/ci/sources/tunnel.sh
 
+# Check if tests can be skipped
+skip_tests=true
+for filename in $(git diff --name-only $TRAVIS_BRANCH...HEAD); do
```

Comment:
I see what you meant. You are right, for Push build, the ``diff`` would be empty.
I should check the ``$TRAVIS_PULL_REQUEST`` first. 

And I did a little search: 
https://github.com/GoogleCloudPlatform/python-docs-samples/blob/72fa14b3957cacfdfe3113d14b8d77204f7ec9f7/nox.py#L84